sharedbuffers/mapped_struct.py

```diff
@@ -1760,6 +1823,255 @@ def _c_search_hkey_ui32(hkey, pindex, stride0, length, hint):
                 return mid
         return lo
 
+    @cython.cfunc
+    @cython.locals(
+        hkey = cython.longlong,
+        mkey = cython.longlong, ikey = cython.int, imkey = cython.int,
+        lo = cython.size_t, hi = cython.size_t, length = cython.size_t,
+        mid = cython.size_t, mid2 = cython.size_t, stride0 = cython.size_t, hint = cython.size_t,
+        pindex = cython.p_char, skip = cython.size_t)
+    @cython.returns(cython.size_t)
+    def _c_search_hkey_i32(hkey, pindex, stride0, length, hint):
+        hi = length
+        lo = 0
+        ikey = hkey
+        if lo < hi:
+            mid = min(hint, hi-1)
+            imkey = cython.cast(cython.p_int, pindex + stride0 * mid)[0]
```

Comment:
No. It's the `i32` type, which means the array is an array of 32-bit ints. That means `p_int`.